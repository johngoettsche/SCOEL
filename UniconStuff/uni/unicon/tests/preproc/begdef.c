/* tests of #begdef - #enddef */

/*
 * object-like macros
 */
#begdef empty
#enddef

#begdef one
1
#enddef

#begdef many_lines
testing
multi-line
macro with
several
lines
#enddef

>empty<
>one<
>many_lines<

/*
 * function-like macros
 */
#begdef nothing()
#enddef

#begdef something()
"everything"
#enddef

#begdef echo(x)
x
#enddef

#begdef abc(a, b, c)
a
-
b
-
c
#enddef

>nothing()<   >something()<   >echo("hi")<    >abc((1,1),2,two)<
