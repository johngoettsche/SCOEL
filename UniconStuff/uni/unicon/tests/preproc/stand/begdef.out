#line 20 "begdef.c"

><
>

#line 9 "begdef.c"

1

#line 21 "begdef.c"

<
>

#line 13 "begdef.c"

testing
multi-line
macro with
several
lines

#line 22 "begdef.c"

<

 
#line 46 "begdef.c"

><   >

#line 31 "begdef.c"

"everything"

#line 46 "begdef.c"

<   >

#line 35 "begdef.c"

"hi"

#line 46 "begdef.c"

<    >

#line 39 "begdef.c"

(1,1)
-
2
-
two


<
